contents/android/IntroToAndroid.md

```diff
@@ -0,0 +1,108 @@
+
+#### How does this link to MVVM?
+Previously, Android apps were usually developed using the `Model-View-Presenter` (MVP) pattern, which is a derivative from the popular Model-View-Controller (MVC) pattern. However, this will change with the above `LiveData` and `ViewModel` components. Now, we can create apps that follow the `Model-View-ViewModel` (MVVM) architecture instead. The `ViewModel` component provided by Android serves as a good base for our ViewModel in the MVVM architecture. Our ViewModel in MVVM will interact with the Model and manages and store data to be observed by a View. The ViewModel is not aware about the View that is getting data from it to update the UI and thus it is decoupled from the View. This is very unlike the Presenter in the MVP pattern, which directly manipulates the View. To read more about the MVVM architecture in Android apps, see this [tutorial](https://proandroiddev.com/mvvm-architecture-viewmodel-and-livedata-part-1-604f50cda1).
+
+
+### 2.5 Testing
+Test-driven development was [encouraged](https://www.youtube.com/watch?v=pK7W5npkhho) by Google in their Google I/O 2017 due to the release of the `Android Testing Support Library`. You can read more about the library [here](https://developer.android.com/topic/libraries/testing-support-library/index.html).
+
+#### Unit Testing and Mocking
+Mocking is usually needed in unit testing to simulate the behaviour of real objects (that are dependencies) to verify the behaviour of the object that you are testing. In Android app development, [Mockito](http://site.mockito.org/), a popular mocking library for unit tests in Java, is often used. However, due to the difficulty of mocking the Android SDK to create unit tests, [Robolectric](http://robolectric.org/) was developed to solve this issue. Robolectric can still be used alongside Mockito. The benefit of Robolectric is that it handles emulation UI code such that tests that rely on the UI can be run on the Java Virtual Machine (JVM) rather than having to run them on an emulator. In other words, tests can be written and run faster.
+
+#### Integration Testing
+Integration testing and end-to-end testing is usually done using [Espresso](https://developer.android.com/training/testing/espresso/index.html). You can even record your own UI tests using the [Espresso Test Recorder](https://developer.android.com/studio/test/espresso-test-recorder.html) to save time! Be sure to check out the Espresso [cheatsheet](https://developer.android.com/training/testing/espresso/cheat-sheet.html) for easy writing of tests.
+
+
+## 3. Further Readings
+Going further, it will be useful to know about:
```

Comment:
Might want to give a very brief description to how these technologies link to android?